cmake/Codegen.cmake

```diff
@@ -208,11 +208,19 @@ endif()
 
 function(get_filelist name outputvar)
   set(_rootdir "${CMAKE_CURRENT_LIST_DIR}/../")
+  # configure_file adds its input to the list of CMAKE_RERUN dependencies
+  configure_file(
+      ${CMAKE_CURRENT_LIST_DIR}/../tools/build_variables.bzl
+      ${CMAKE_BINARY_DIR}/caffe2/build_variables.bzl)
```

Comment:
I'm confused. the rest of this function only uses `../tools/build_variables.bzl`, so why is `${CMAKE_BINARY_DIR}/caffe2/build_variables.bzl` involved and where does it come from?

Reply:
never mind, didn't realize configure_file copies the file